lab_1_keywords_tfidf/main.py

```diff
@@ -114,6 +145,19 @@ def get_top_n(frequencies: dict[str, int | float], top: int) -> list[str] | None
         list[str] | None: Top-N tokens sorted by frequency.
         In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(frequencies, dict):
+        return None
+    if not isinstance(top, int) or isinstance(top, bool) or top <= 0:
+        return None
+    if not frequencies:
+        return None
+    for key, value in frequencies.items():
+        if not isinstance(key, str) or not isinstance(value, (int, float)):
+            return None
+    sorted_words = sorted(frequencies.items(), key=lambda x: x[1], reverse=True)
+    if top >= len(sorted_words):
+        return [word for word, freq in sorted_words]
+    return [word for word, freq in sorted_words[:top]]
```

Comment:
If you sort the keys of `frequencies` dict, you can avoid additional iteration through list and just take a slice with required number of tokens